### PR TITLE
Used Mage_Core_Model_App_Area constants instead of string literals

### PR DESCRIPTION
**Pull request by Sdfendor**
The app area model provides constants for all the common areas. In the codebase there are places where string literals are used nontheless. The PR replaces mainly 'adminhtml' and 'frontend' string literals with the corresponding constants where it is appropriate (Both strings are used in other contexts as well). 
Some of the lines are reformatted to match their surrounding lines and exactly two string comparisons are changed from loose to strict.

### CODE REVIEW

File: app/code/core/Mage/Directory/Model/Observer.php [hunk 1]
```diff
@@ -65,17 +65,21 @@ public function scheduledUpdateCurrencyRates()
...
-                [
-                    'warnings' => implode("\n", $importWarnings),
-                ]
-            );
+            $mailTemplate
+                ->setDesignConfig(
+                    [
+                        'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND
+                    ]
+                )
```
Comment (fballiano) [inline app/code/core/Mage/Directory/Model/Observer.php:73]:
```suggestion
                ->setDesignConfig([
                        'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND
                ])
```

File: app/code/core/Mage/ProductAlert/Model/Email.php [hunk 2]
```diff
@@ -261,10 +261,13 @@ public function send()
...
-            ->setDesignConfig([
-                'area'  => 'frontend',
-                'store' => $storeId
-            ])->sendTransactional(
+            ->setDesignConfig(
+                [
+                    'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND,
+                    'store' => $storeId
+                ]
+            )
```
Comment (fballiano) [inline app/code/core/Mage/ProductAlert/Model/Email.php:269]:
```suggestion
            ->setDesignConfig([
                    'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND,
                    'store' => $storeId
            ])
```

**Comment by fballiano** [comment]
hey @Sdfendor thanks a lot for this PR, a like it a lot!

@justinbeaty do you approve it?

**Review by justinbeaty — APPROVED** [review]
On a flight now so only a quick review, but it's always good to get rid of hardcoded strings so definitely a good idea!

**Comment by Sdfendor** [comment]
@fballiano I applied the shorter format when calling a method with only one array argument, following your suggestions.

**Comment by fballiano** [comment]
thanks everybody ❤️

@Sdfendor do you need to change something else or shall I merge it right away?

**Comment by Sdfendor** [comment]
Oh I guess I added some whitespace where it shouldn't be, the linters failed.

File: app/code/core/Mage/Directory/Model/Observer.php [hunk 3]
```diff
@@ -65,17 +65,19 @@ public function scheduledUpdateCurrencyRates()
...
-                Mage::getStoreConfig(self::XML_PATH_ERROR_IDENTITY),
-                Mage::getStoreConfig(self::XML_PATH_ERROR_RECIPIENT),
-                null,
-                [
-                    'warnings' => implode("\n", $importWarnings),
-                ]
-            );
+            $mailTemplate
+                ->setDesignConfig([
+                    'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND
```
Comment (fballiano) [inline app/code/core/Mage/Directory/Model/Observer.php:70]:
```suggestion
                    'area' => Mage_Core_Model_App_Area::AREA_FRONTEND
```

File: app/code/core/Mage/Checkout/Helper/Data.php [hunk 4]
```diff
@@ -230,7 +230,11 @@ public function sendPaymentFailedEmail($checkout, $message, $checkoutType = 'one
         $total = $checkout->getStoreCurrencyCode() . ' ' . $checkout->getGrandTotal();
 
         foreach ($sendTo as $recipient) {
-            $mailTemplate->setDesignConfig(['area' => 'frontend', 'store' => $checkout->getStoreId()])
+            $mailTemplate
+                ->setDesignConfig([
+                    'area' => Mage_Core_Model_App_Area::AREA_FRONTEND,
```
Comment (fballiano) [inline app/code/core/Mage/Checkout/Helper/Data.php:235]:
```suggestion
                    'area'  => Mage_Core_Model_App_Area::AREA_FRONTEND,
```

Comment (Sdfendor) [reply app/code/core/Mage/Checkout/Helper/Data.php]: Ah yes I wasn't used to the "align the =>" form. Sorry for that 😅.

**Comment by fballiano** [comment]
sorry for the silly changes but now I think everything looks ok, when workflows finish I'll merge :-)

**Comment by Sdfendor** [comment]
No problem. Just getting used to the aligning of array elements 😉.

**Pull request merged**